File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/repository/ITestParticipantRelationRepository.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface ITestParticipantRelationRepository <ID, T> : IRepository<ID, T> where T : HasId<ID>
    {
        IEnumerable<int> findAllParticipantsIDForTest(int id);
    }
}

File: Java_ChildrenCompetitionSockets/CompetitionPersistence/src/main/java/competition/persistence/ITestParticipantRelationRepository.java
package competition.persistence;


//public interface ITestParticipantRelationRepository<ID, T extends HasId<ID>> extends IRepository<ID, T> {
//
//}

import competition.model.TestParticipantRelation;
import competition.model.Tuple;

public interface ITestParticipantRelationRepository extends IRepository<Tuple<Integer, Integer>, TestParticipantRelation> {

}